go/internal/feast/errors.go

```diff
@@ -0,0 +1,21 @@
+package feast
+
+import (
+	"fmt"
+	"strings"
+)
+
+// TODO: Add more errors here especially once that have been created in python sdk
```

Comment:
It seems like this file is meant to duplicate what we did in Python. I don't really see the value yet. NewFeatureNameCollisionError is used in a single place. Can we move this error handling logic all into featurestore.go until we absolutely need to generalize across files?